### PR TITLE
PMM-9362 Force disconnect

### PR DESCRIPTION
**Pull request by oksana-grishchenko**
PMM-9362 Force disconnect 

If for some reason a PMM can not obtain a user token (for default PMM admin user or when the SSO is broken), the PMM admin should be able to cleanup the PMM side of the connection - remove sso_details record and update supervisord configurations. 

Build: SUBMODULES-0

- [ ] https://github.com/percona/pmm/pull/876

### CODE REVIEW

**Comment by oksana-grishchenko** [comment]
`CI / Test (1.18.x, ubuntu-latest, false) ` failure seems to be unrelated, I can also see it in other PRs in this repo

File: services/platform/platform.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (s *Service) Disconnect(ctx context.Context, req *platformpb.DisconnectRequ
 	err = s.disconnect(ctx, &disconnectPMMParams{
 		PMMServerID: settings.PMMServerID,
 	})
-	if err != nil {
+	if err != nil && !req.Force {
```
Comment (ritbl) [inline services/platform/platform.go:172]: probably it should be `|| req.Force`?

**Review by BupycHuk — COMMENTED** [review]
BTW why don't we use transaction in disconnect?

File: services/platform/platform.go (hunk 1 above)
Comment (oksana-grishchenko) [reply services/platform/platform.go:172]: I don't think so. This block restores the above deleted data.
In case of an error we need to restore it only when the request is not forced.
In case of `|| req.Force` the data will be restored even when there was no error at all, which is wrong. 

I agree it's hard to follow, but it's how the method is designed.

Comment (BupycHuk) [reply services/platform/platform.go:172]: I suggest to use transaction in `disconnect`, so it won't be so hard to follow.

Comment (ritbl) [reply services/platform/platform.go:172]: oh... I see. but that is super unclear. How about a bit of refactoring?

we can have:
1: extra variable:
```
recover := err != nil && !req.Force
if recover {
...
}
```

2: extract method:
```
if err != nil && !req.Force {
 recoverXyz() // here we need a good name, as it the scope grows
}
```

Comment (BupycHuk) [reply services/platform/platform.go:172]: Please check usage of transaction here https://github.com/percona/pmm-managed/blob/main/services/management/mysql.go#L58
It rollbacks changes in case of error.

**Review by artemgavrilov — APPROVED** [review]

**Comment by artemgavrilov** [comment]
> BTW why don't we use transaction in disconnect?

I guess it's because disconnect is blocking network operation, and we don't wan't to hang transaction because of network problems.

File: services/platform/platform.go (hunk 1 above)
Comment (BupycHuk) [reply services/platform/platform.go]: ok, sounds reasonable.

**Review by palash25 — APPROVED** [review]
